src/Knp/Component/Pager/ParametersResolver/Symfony.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Knp\Component\Pager\ParametersResolver;
+
+use Knp\Component\Pager\ParametersResolver;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class Symfony implements ParametersResolver
+{
+    private $request;
+
+    public function __construct(RequestStack $requestStack)
+    {
+        $this->request = $requestStack->getCurrentRequest();
+    }
+
+    public function get(string $parameterName, ?string $defaultValue): ?string
+    {
+        return $this->request->get($parameterName, $defaultValue);
```

Comment:
You know what I would like to see here? Before fallback to default value, fallback to route default parameter first. I'm dealing with that issue right now. Currently this needs to be solved in userspace with dirty hacks.

```php
$defaultRouteParams = $request->attributes->get('_route_params', []);

return $this->request->get($parameterName, $defaultRouteParams[$parameterName] ?? $defaultValue);
```